solr/core/src/java/org/apache/solr/util/tracing/HttpServletCarrier.java

```diff
@@ -29,9 +29,11 @@
  * A Carrier for extract Span context out of request headers
  */
 public class HttpServletCarrier implements TextMap {
+  private final HttpServletRequest req;
   private Iterator<Map.Entry<String, String>> it;
 
   public HttpServletCarrier(HttpServletRequest request) {
+    this.req = request;
```

Comment:
we are holding request object here. how this 'HttpServletCarrier' object works? it per request?

Reply:
It's a short lived object that is used in a method call. This object is not stored anywhere